src/lib/storage/frame_of_reference_segment/frame_of_reference_segment_iterable.hpp

```diff
@@ -142,7 +142,7 @@ class FrameOfReferenceSegmentIterable : public PointAccessibleSegmentIterable<Fr
 
    private:
     const pmr_vector<T>* _block_minima;
-    const pmr_vector<bool>* _null_values;
+    const std::optional<pmr_vector<bool>>* _null_values;
```

Comment:
Pointers already have "does not exist" semantics, namely nullptr. Using that would allow us to avoid `**_null_values` above.

Reply:
Personally, I'd favor std::optional as it clearly states the intend. `nullptr` are bad (IMO). I could measure if `*(_null_values.get())` does have a negative impact if the double star is of concern?

Reply:
Mmmh, I can't find a good canonical answer for that. To me, it opens up more ambiguities. Why is this an `optional<T>*`, i.e., a pointer to an optionally existing value, and not an `optional<T*>`, meaning an optionally existing pointer to a certainly existing value.

Also, having `if (_null_values)` (which always returns true), `if (*_null_values)` (which might return true), and `if ((**_null_values)[x])` (which depends on the data) seems unnecessarily complicated. For funsies, we could even add `if (***_null_values)` to the discussion, which checks if the first value is NULL. ;)

In the great scheme of things, no one will get physically hurt by this, so whatever...

Reply:
I see your point, but the current way of `optional<T>*` mirrors exactly what it done to all other members of the segment as well. Also, I think "an optionally existing pointer to a certainly existing value" is more confusing. But that might just be me.